pyzeebe/worker/job_executor.py

```diff
@@ -35,6 +36,9 @@ async def get_next_job(self) -> Job:
     async def execute_one_job(self, job: Job, job_controller: JobController) -> None:
         try:
             await self.task.job_handler(job, job_controller)
+        except NonRecoverableDecoratorError as error:
+            logger.error(f"Non-recoverable decorator error in job %s: %s", job.type, error)
+            raise
```

Comment:
I think it makes more sense to put this code in `pyzeebe/task/task_builder.py` inside `run_decorator` function where you catch this exception. Also, does the rest of the code treat this exception as job failure without it being a subclass of `PyZeebeError`?